### PR TITLE
Per-rule filters now actually work!!

### PR DESCRIPTION
**Pull request by walaura**
This also does a couple UI niceties, namely:

- Popovers now try to fit in the screen and will scroll as a worst case scenario
- Progress bars have nicer stripes
- The filter status text has the colour of the selected rule
- Clicking on the empty progress bar space shows no matches

And code niceties:

- helpers are in a new `helpers` directory instead of alongside modules

screenshot 2018-12-05 at 12 24 57 pm
screenshot 2018-12-05 at 12 25 01 pm

### CODE REVIEW

File: jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (walaura) [inline jsconfig.json]: this lets vs code click through imports!

Comment (itsibitzi) [reply jsconfig.json]: Amaaaazzinnngggg

File: src/views/Project/Project.js [hunk 2]
```diff
@@ -141,21 +151,70 @@ class Project extends Component {
 						}}
 						label={'Add Dataset'}
 					/>
-					<MenuItem type="separator" />
+					{project.datasets.length && <MenuItem type="separator" />}
```
Comment (itsibitzi) [inline src/views/Project/Project.js]: Have you checked this renders ok with the zero value? I've had some issues with react rendering the number zero. Can change it to `!!project.datasets.length` if it doesn't work.

Comment (walaura) [reply src/views/Project/Project.js]: this one's purpose, i only want the line if there's `> 0`! does it read confusingly? (I wouldn't worry too much bc this dataset UI is temporary filler)

**Pull request merged**